### PR TITLE
Quiz

### PR DESCRIPTION
**Pull request by botondhegedus**
quiz.js

### CODE REVIEW

File: lib/quiz.js [hunk 1]
```diff
@@ -6,8 +6,108 @@ var restrict_clause = require('./sql_restrictions.js')(),
...
+                "WHERE log.module = 'quiz'" + 
+                "AND log.action = 'add'" + 
+                "AND " + restrict_clause,
+
+        sql_old_2pass: (row) => {
+            var m = row.url.match(/id=(\d+)/);
+            if (!m){
+                console.log('No course module ID in URL ' + JSON.stringify(row));
+                return null;
+            }
```
Comment (jlabusch) [inline lib/quiz.js]: This looks unfinished; what's the query you want to execute if id does exist?

File: lib/quiz.js [hunk 2]
```diff
@@ -6,8 +6,108 @@ var restrict_clause = require('./sql_restrictions.js')(),
...
+                    'WHERE m.name = ? AND c.shortname = ? AND q.name = ?',
+                    [
+                        row["u_email"],
+                        row["u_username"],
+                        'quiz',
+                        row["course_shortname"],
+                        row["quiz_name"],
+                    ]
+                )
+                :
```
Comment (jlabusch) [inline lib/quiz.js:55]: See comment from other PR - the ternary operator isn't being used correctly here.

File: lib/quiz.js [hunk 3]
```diff
@@ -6,8 +6,108 @@ var restrict_clause = require('./sql_restrictions.js')(),
...
+                console.log('No course module ID from DB ' + JSON.stringify(row));
```
Comment (jlabusch) [inline lib/quiz.js]: This is redundant; just return null and the caller will print the row for you with an error message.

**Review by jlabusch — CHANGES_REQUESTED** [review]
functions for sql_old_2pass and sql_match have similar errors throughout the file, possibly caused by copy-paste.

File: lib/quiz.js (hunk 2 above)
Comment (botondhegedus) [reply lib/quiz.js:55]: Fixed

**Comment by botondhegedus** [comment]
I made some changes, ready to test

**Review by jlabusch — APPROVED** [review]

**Pull request merged**